custom_components/elasticsearch/entity_details.py

```diff
@@ -28,27 +38,77 @@ def __init__(self, hass: HomeAssistant):
         self._registry_entry = entity_registry.async_get(self._hass)
         self._registry_device = device_registry.async_get(self._hass)
         self._registry_area = area_registry.async_get(self._hass)
+        self._registry_floor = floor_registry.async_get(self._hass)
+        self._registry_label = label_registry.async_get(self._hass)
 
     def async_get(self, entity_id: str) -> FullEntityDetails | None:
         """Retrieve entity details."""
 
-        entity: entity_registry.RegistryEntry = self._registry_entry.async_get(entity_id)
+        # Use helper functions below to assemble a FullEntityDetails object
 
+        # Get the entity
+        entity = self.async_get_entity(entity_id)
         if entity is None:
             LOGGER.debug("Entity not found: %s", entity_id)
             return None
 
-        entity_area: area_registry.AreaEntry = None
-        if entity.area_id is not None:
-            entity_area = self._registry_area.async_get_area(entity.area_id)
+        entity_label = self.async_get_entity_label(entity)
+        entity_area = self.async_get_entity_area(entity)
+
+        # Get the device of the entity
+        device = self.async_get_device(entity.device_id)
+        if device is None:
+            LOGGER.debug("Device not found for entity: %s", entity_id)
+
+        device_label = self.async_get_device_label(device)
+        device_area = self.async_get_device_area(device)
+        device_floor = self.async_get_area_floor(device_area)
+
+        return FullEntityDetails(
+            entity,
+            entity_area,
+            entity_label,
+            device,
+            device_area,
+            device_floor,
+            device_label,
+        )
+
+    # Entity functions
+    def async_get_entity(self, entity_id: str) -> entity_registry.RegistryEntry | None:
+        """Retrieve entity details."""
+        return self._registry_entry.async_get(entity_id)
+
+    def async_get_entity_area(self, entity) -> area_registry.AreaEntry | None:
+        """Retrieve entity area details."""
+        if entity.area_id is None:
+            return None
+        return self._registry_area.async_get_area(entity.area_id)
 
-        device: device_registry.DeviceEntry = None
-        device_area: area_registry.AreaEntry = None
-        if entity.device_id is not None:
-            device = self._registry_device.async_get(entity.device_id)
-            if device.area_id:
-                device_area = self._registry_area.async_get_area(device.area_id)
+    def async_get_entity_label(self, entity) -> label_registry.LabelEntry | None:
```

Comment:
This type definition looks incorrect. This is always returning a list of entries, not an optional single entry.

Also suggest updating the function name to indicate a collection instead of a single entry:


```suggestion
    def async_get_entity_labels(self, entity: entity_registry.RegistryEntry) -> list[label_registry.LabelEntry]:
```